community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -215,6 +221,8 @@ public KernelTransactionImplementation initialize(
         this.transactionEvent = tracer.beginTransaction();
         assert transactionEvent != null : "transactionEvent was null!";
         this.accessMode = accessMode;
+        this.transactionId = NOT_COMMITTED_TRANSACTION_ID;
+        this.commitTime = NOT_COMMITTED_TRANSACTION_COMMIT_TIME;
```

Comment:
I'm not sure what race your talking about, but let me give some explanation to start with :)

Newly introduced `transactionId` and `commitTime` introduced for and currently used only by transaction event handlers and they can't race with anyone for their transaction.

But what lets assume malicious user will try to access that data using `KernelTransactions.activeTransactions`?  That call do not expose `KernelTransaction` anymore for external usage, but use newly introduced `KernelTransactionImplementationHandle` so reuse of transaction in regards to newly introduced fields should be safe.
If you still have some questions or have a case in mind please shoot!